### PR TITLE
Optimize CI by invoking clang tidy only on modified files

### PR DESCRIPTION
**Pull request by sebalaig**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#173 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,12 +24,19 @@ jobs:
...
-          -DCMAKE_CXX_CLANG_TIDY=clang-tidy
           -DCMAKE_PREFIX_PATH=$BOOST_ROOT_1_72_0
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
 
       - name: Build
         run: cmake --build $GITHUB_WORKSPACE/build-linux --parallel 3
 
+      - name: Clang Tidy
+        run: |
+          export MODIFIED_FILES=$(git diff -U0 HEAD^ --name-only |grep -E "\.(cpp|hpp|hxx)$")
```
Comment (mathbagu) [inline .github/workflows/ci.yml:35]: Do not export the variable and add a space between `|` and `grep`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,12 +24,19 @@ jobs:
...
+          if [ -n "$MODIFIED_FILES" ]; then
+              clang-tidy $MODIFIED_FILES -p $GITHUB_WORKSPACE/build-linux
+          fi
+
```
Comment (mathbagu) [inline .github/workflows/ci.yml:39]: I think clang-tidy check should be moved to the `QA` job. Don't forget to remove the installation of clang-tidy from the linux job too.

**Review by mathbagu — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sebalaig) [reply .github/workflows/ci.yml]: Without exporting variable, CI failed : https://github.com/powsybl/powsybl-iidm4cpp/actions/runs/244562520 :/
Maybe I missed something ?

Comment (mathbagu) [reply .github/workflows/ci.yml]: I don't understand why there is an error. On my Fedora, without the `export`, it works properly. We keep it like that

**Pull request merged**